src/main/kotlin/study/StringAddCalculator.kt

```diff
@@ -0,0 +1,25 @@
+package study
+
+class StringAddCalculator {
+    fun add(input: String?): Int {
+        if (input.isNullOrBlank()) {
+            return 0
+        }
+
+        // 커스텀 구분자 있는지 확인
+        val result = Regex("//(.)\n(.*)").find(input)
+        return result?.let {
+            val customDelimiter = it.groupValues[1]
+            val tokens = it.groupValues[2].split(customDelimiter)
+            return@let add(tokens)
+        } ?: add(input.split(",|:".toRegex()))
+    }
+    
+    private fun add(numbers: List<String>): Int {
+        return numbers.map { it.toInt() }
+            .filter {
+                if (it > -1) true
+                else throw RuntimeException() }
```

Comment:
아래 처럼 정렬해보는 건 어떨까요?
```suggestion
            .filter {
                if (it > -1) true
                else throw RuntimeException() 
            }
```

또, Exception에는 어떤 이유에 발생했는지 인풋 값과 함께 메시지를 남겨보는 건 어떨까요?